### ISSUE TITLE
Make Expr.head(), Series.head(), DataFrame.head() and LazyFrame.head() return the same number of rows by default

### ISSUE THREAD

**Issue by Wainberg** [pola-rs/polars#13445]
### Description

Expr.head() and Series.head() return 10 rows by default, while DataFrame.head() and LazyFrame.head() return 5 rows. We should pick one ;)

**Comment by mcrumiller** [comment]
I like 5. The point is to see what the table looks like (columns, dtypes, reasonable samples of data). and adding more rows than that might make you scroll up in a smallish terminal.

**Comment by stinodego** [comment]
I don't think we need to pick one at all. One is a single column while the other is a 2D table. 10 rows of a dataframe is a lot of information, 10 elements of a column not so much.

There is also something to be said for having the same value across data types, but it's not so black and white as you make it out to be.

**Comment by Wainberg** [comment]
I would argue for either using 5 for all, or changing the default number of printed rows. Currently you can't see the intermediate rows with Expr.head():

```python
>>> pl.DataFrame(range(100)).select(pl.all().head())
shape: (10, 1)
┌──────────┐
│ column_0 │
│ ---      │
│ i64      │
╞══════════╡
│ 0        │
│ 1        │
│ 2        │
│ 3        │
│ …        │
│ 6        │
│ 7        │
│ 8        │
│ 9        │
└──────────┘
```

**Comment by Wainberg** [comment]
Would you be open to changing the default number of printed rows from 8 to 10?

**Comment by alexander-beedie** [comment]
> Would you be open to changing the default number of printed rows from 8 to 10?

Apparently so 😎

**Comment by Wainberg** [comment]
Love it!!

Ok so at this point I would definitely advocate for `head()`, `tail()` and `limit()` to return 10 rows instead of 5. I'm sympathetic to Stijn's argument that 10 rows of a DataFrame can be more of an 'information overload' than 5 rows, but if we're now printing 10 rows of a DataFrame by default, may as well do the same for `head()` etc.

**Comment by alexander-beedie** [comment]
> if we're now printing 10 rows of a DataFrame by default, may as well do the same for `head()` etc.

Counter-argument: in those 10 rows we're displaying the equivalent of `head(5)` and `tail(5)` once the total number of rows >= 10 😉

**Comment by mcrumiller** [comment]
@alexander-beedie very good point. No matter what `n` we pick for `head` and `tail`, we must show `2n+1` or `2n-1` rows in `df.__str__()` without an asymmetric head/tail.

**Comment by mcrumiller** [comment]
My vote here is:

* 10 rows for `df.head()` and `df.tail()`
* 10 rows for `df.limit()`
* first 5 and last 5 for `df.__str__()`
